File: ios/CoffeeBreak/Models/GameElement.swift
import Foundation

/// The 6 matchable elements (pâtisserie theme), per 03-assets-de-jeu.md.
/// Match-3 board mechanics land with 05-mecaniques-de-jeu.md.
enum GameElement: String, CaseIterable {
    case cupcakeGlacageChocolat
    case donutGlacageRose
    case tarteletteFraiseChantilly
    case croissant
    case partGateauCouches
    case bretzel

    /// Matches the imageset name in Assets.xcassets/GameElements.
    var assetName: String {
        switch self {
        case .cupcakeGlacageChocolat: return "element-cupcake-glacage-chocolat"
        case .donutGlacageRose: return "element-donut-glacage-rose"
        case .tarteletteFraiseChantilly: return "element-tartelette-fraise-chantilly"
        case .croissant: return "element-croissant"
        case .partGateauCouches: return "element-part-gateau-couches"
        case .bretzel: return "element-bretzel"
        }
    }
}

File: mobile/lib/models/game_element.dart
/// The 6 matchable elements, per 03-assets-de-jeu.md.
enum GameElement {
  cupcake,
  donut,
  tartelette,
  croissant,
  partGateau,
  bretzel;

  String get assetName {
    switch (this) {
      case GameElement.cupcake:
        return 'assets/game_elements/element-cupcake-glacage-chocolat.png';
      case GameElement.donut:
        return 'assets/game_elements/element-donut-glacage-rose.png';
      case GameElement.tartelette:
        return 'assets/game_elements/element-tartelette-fraise-chantilly.png';
      case GameElement.croissant:
        return 'assets/game_elements/element-croissant.png';
      case GameElement.partGateau:
        return 'assets/game_elements/element-part-gateau-couches.png';
      case GameElement.bretzel:
        return 'assets/game_elements/element-bretzel.png';
    }
  }
}
